### PR TITLE
[doc] Version Control Exercise

### PR DESCRIPTION
**Pull request by zwgold**
resolves umrover/mrover-ros#41

Adapted the EECS 280 Lab 0 Version Control Exercises into a markdown for new members to work through to gain experience handling changes across different local repositories.

### CODE REVIEW

File: Version_Control_Exercise.md
Comment (tabiosg) [inline Version_Control_Exercise.md]: References to GitLab can be deleted

**Review by tabiosg — COMMENTED** [review]
The exercise has useful information. I like it. Just delete references to GitLab.

**Review by niwhsa9 — APPROVED** [review]
Good stuff, nice work. LGTM pending resolving what @tabiosg pointed out

**Comment by zwgold** [comment]
Do you guys think it'd be a good idea to add some diagrams to how the commits/workflow occurs? I'm fine either way, but what we already have is a lot so I understand if not

**Comment by tabiosg** [comment]
> Do you guys think it'd be a good idea to add some diagrams to how the commits/workflow occurs? I'm fine either way, but what we already have is a lot so I understand if not

I personally never found those diagrams useful when I was trying to learn Git (because they are usually too "complicated" for the basics). 

The most useful image I've ever came across about was this one since it helped me understand the difference between local and remote:

here

Having diagrams for merging and pull requests is quite messy and complicated, and if you do add a diagram for that, I would keep it to one (I was looking for a good diagram as an example but gave up since all of them would probably just create confusion).

**Comment by zwgold** [comment]
Honestly, at that point, I think we'd be fine to leave it out. I think for the wiki it may be a good idea for one of us to make some illustration or diagram of how MRover workflow works since this is merely a tutorial/intro. I think leaving it as is without a diagram is fine.

**Comment by niwhsa9** [comment]
I imagine some whiteboarding sessions during meetings will still be in order, even when we front load the teaching effort with great resources like this. I hope everyone has been practicing drawing git trees :laughing:

**Comment by zwgold** [comment]
Ok, last set of changes. Mostly grammatical, some smaller things with boldening and me leaving in "lab staff." I also added a note that rebasing should not be an option almost never. Feel free to peek quickly and once @CameronTressler reviews and approves, we can merge

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Maybe it would be good to explain the 'vc' acronym, people may not realize what this is. Within this, could say that svn is an alternative to git

Comment (CameronTressler) [reply Version_Control_Exercise.md]: Alternatively I guess I could talk about this in the workflow-intros readme, but I think it would make a little more sense here? That other one is really just about pull requests

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Could this be changed to describe how to find the link from the browser page? Personally, I very rarely just type the link directly, usually I copy it from the 'code' drop down

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Might be a little pedantic but I would recommend explicitly saying 'using `cd mrover-vs-exercise`'

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Looks like a newline accidentally got into this line

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Just for clarity, I would say "This shows that the file is now staged..." to be clear that git status doesn't do anything

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Same comment as above

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: I'm tempted to just say we should describe what 'upstream' here means, but that might be pretty challenging for a novice git user to understand, and not completely necessary for now. Alternatively, we could just tell them to run `git push origin main`

File: Version_Control_Exercise.md
Comment (CameronTressler) [inline Version_Control_Exercise.md]: Hopefully they get the point, but just in case, probably good to specify this should happen before pulling

**Comment by CameronTressler** [comment]
Looks good! I added a few small comments that should be pretty straight-forward to address. Sorry for taking a few days

File: Version_Control_Exercise.md
Comment (zwgold) [reply Version_Control_Exercise.md]: I'll put something in the beginning that talks about VCS and Git vs. SVM; I didn't feel like it would be necessary, but it would be good for members to know

File: Version_Control_Exercise.md
Comment (zwgold) [reply Version_Control_Exercise.md]: I thought about it, I'm going to change the command, but additionally, explain what upstream means. I'll send what I had in mind and if it's too much we can just use the command `git push origin main` with a very brief explanation

File: Version_Control_Exercise.md
Comment (zwgold) [reply Version_Control_Exercise.md]: Good call

**Review by tabiosg — APPROVED** [review]

File: Version_Control_Exercise.md
Comment (CameronTressler) [reply Version_Control_Exercise.md]: sounds good

File: Version_Control_Exercise.md
Comment (CameronTressler) [reply Version_Control_Exercise.md]: I think this is good

**Review by CameronTressler — APPROVED** [review]

**Pull request merged**
